### DISCUSSION TITLE
How to implement data fetchers for nested mutation types

### DISCUSSION THREAD

**Discussion by sriramgopalan-awx** [Netflix/dgs-framework#1444] [Q&A]
```
type Mutation {
    topLevel: MyTopLevelMutation!
}

type MyTopLevelMutation {
 
    signUp(input: UserInput!): UserPayload!
    
    #...more mutations
}

type UserInput {
   first: String!
   last: String!
}

type UserPayload {
    success: Boolean!
    id: ID!
}

```
As you see in the above schema, I have a nested mutation structure and the operations are not at the root level. Now how do I implement data fetchers for `topLevel` which is of type mutation and for `signUp` field which is of  type `MyTopLevelMutation` ? Any help will be really appreciated.

https://github.com/Netflix/dgs-framework/discussions/528 - In this issue they refer to testing but I would like to know the implementation of the data fetchers.

**Comment by justincohan** [comment]
I had to do something like this example where I have a mutation that returns a class with a reference to the nested payload which will be populated by the function annotated with @DgsData

```
@DgsComponent
class MyResolver() {

    data class MyTopLevelMutation(
        var userPayload: UserPayload?
    )

    @DgsMutation
    fun myTopLevelMutation(): MyTopLevelMutation {
        return MyTopLevelMutation(null)
    }

    @DgsData(parentType = "MyTopLevelMutation", field = "topLevel")
    fun topLevel(@InputArgument input: UserInput): UserPayload {
        return UserPayload()
    }
}
```
